packages/flutter_tools/lib/src/commands/run.dart

```diff
@@ -592,6 +593,11 @@ class RunCommand extends RunCommandBase {
     if (devices == null) {
       throwToolExit(null);
     }
+
+    if (devices!.length == 1 && devices!.first is MacOSDesignedForIPadDevice) {
```

Comment:
I think you want to fail if any of them are `MacOSDesignedForIPadDevice`, not just if there's one.
```suggestion
    if (devices!.any((Device device) => device is MacOSDesignedForIPadDevice)) {
```

Reply:
oh yeah good point